etcd-recovery/etcdrecovery.go

```diff
@@ -0,0 +1,538 @@
+package etcdrecovery
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"go.uber.org/zap/zapcore"
+
+	hyperapi "github.com/openshift/hypershift/support/api"
+	hyperutil "github.com/openshift/hypershift/support/util"
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
+	etcdclient "go.etcd.io/etcd/client/v3"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	defaultEtcdClientTimeout = 5 * time.Second
+	defaultDialTimeout       = 3 * time.Second
+	healthyClusterWait       = 5 * time.Minute
+)
+
+type options struct {
+	etcdClientCertFile string
+	etcdClientKeyFile  string
+	etcdCAFile         string
+	namespace          string
+}
+
+func NewRecoveryCommand() *cobra.Command {
+	opts := options{
+		etcdClientCertFile: "/etc/etcd/tls/client/etcd-client.crt",
```

Comment:
When would this change so they need to be configurable?

Reply:
It allows testing of the command independently from a local machine (which I had to do many times 😅)